app/containers/filterContainer/filterSaveBox/index.tsx

```diff
@@ -48,40 +54,82 @@ const FilterSaveBox: React.FunctionComponent<FilterSaveBoxProps & RouteComponent
     [props.location, savedFilterSet]
   );
 
-  function handleClickSaveButton(changedFilterReq: Filter | string) {
-    let changedFilter: Filter | null = null;
+  const lastSavedFilters = React.useRef(myFilters);
+
+  React.useEffect(
+    () => {
+      if (currentUserState.isLoggedIn) {
+        if (!isEqual(lastSavedFilters.current, myFilters)) {
+          const newFiltersReq = RequestFilterObjectGenerator(myFilters);
```

Comment:
problems
1. function starts with an upper case.
2. Generator generally means class generator in Javascript Environment.
3. I couldn't get what this function do until I'd checked the content of the function.